### ISSUE TITLE
Test new drawer toggle pattern for a11y

### ISSUE THREAD

**Issue by kimisgold** [omeka/plugin-VraCore#111]
We're revising Omeka Classic's collapsible drawer pattern and applying it to the plugins. This pattern includes components that can collapse and expand. This pattern appears within the item form. You can go to an item's edit page and then the VraCore tab.

Test the following on screen readers for mobile and desktop browsers:

* Each group of collapsible VraCore form fields can expand and collapse.
* Nested drawers are closed by default.
* Nested drawers expand and collapse without affecting their parent drawers and vice versa.
* Toggle buttons announce the field group they control.

You need to be on the `drawer-toggle-a11y-rebase` branch for Omeka Classic, and the `master` branch of VraCore. The core's branch has been updated with fixes impacting nested drawer interactions, so make sure that branch is current.

**Comment by aviannamiller** [comment]
This update overall looks good, and the following are true across the board:
- All form fields can expand and collapse.
- Drawers are automatically closed and stay closed upon collapse until they are expanded again.
- Child and parent drawers expand and collapse without impacting one another.

However, there are some issues with toggle button announcements:
- (VO, TB, NVDA, JAWS) The toggles do not announce the field group name (this can only be accessed via arrow keys, which are read out in sequence first since focus travels there before the toggle). Instead, they just offer a readout of "Show, collapsed (or expanded), button, group" (at least in VO, though JAWS and NVDA have similar announcements). While the toggles themselves do not announce the name on TalkBack or VO on iOS, focus likely travels first to the drawer names, so this is somewhat less of an issue on mobile.
- I am getting some extraneous information depending on if the drawer is expanded or collapsed:
  - (VO, NVDA, JAWS) When the drawer is expanded, the labels of the fields contained within the drawer are read out while focus remains on the toggle (sometimes all are announced, sometimes just some, I have not recognized a clear pattern here yet).
  - (VO) When the drawer is collapsed, content that is in view from the whole grouping of drawers within the container (so beyond the drawer in question) is announced while focus remains on the toggle. To give you a sense of this, in the below example, focus is on the toggle for Display Attributes, but "Notes Show Title Elements Show Add Input Attributes Show type extent" follows the collapsed state announcement.
  <img width="176" alt="Three parent drawers. Focus is on the toggle on the top most drawer. The last drawer is expanded to reveal a child drawer, which is also expanded." src="https://github.com/user-attachments/assets/028e1e97-cfac-452e-87fe-ce1c6814a892" />

  - (VO) If a toggle is activated multiple times, sometimes the above issues reverse, and a collapsed drawer will read out its contents while an expanded drawer will read the contents / names of other drawers. No clear pattern to this, either.

(VO, TB, NVDA, JAWS) I also experienced the toggle become "stuck" in the collapsed or expanded state quite a few times. The only possible cause I could discern for this was that it seemed to occur after experimenting with adding VRA elements.

**Comment by kimisgold** [comment]
> (VO, TB, NVDA, JAWS) The toggles do not announce the field group name (this can only be accessed via arrow keys, which are read out in sequence first since focus travels there before the toggle). Instead, they just offer a readout of "Show, collapsed (or expanded), button, group" (at least in VO, though JAWS and NVDA have similar announcements). While the toggles themselves do not announce the name on TalkBack or VO on iOS, focus likely travels first to the drawer names, so this is somewhat less of an issue on mobile.

I've assigned more group roles and labels throughout the interface to address this.

> I am getting some extraneous information depending on if the drawer is expanded or collapsed:

This is due to an old `aria-live="polite"` markup pattern that no longer exists in the core. This is an older branch that doesn't include the update that fixes this. I'm fairly sure once this gets merged into core, it won't be an issue. I'm going to ask that we let this go for now, but keep it in mind to watch out for after the this pattern is merged in the core and across plugins.

> (VO, TB, NVDA, JAWS) I also experienced the toggle become "stuck" in the collapsed or expanded state quite a few times. The only possible cause I could discern for this was that it seemed to occur after experimenting with adding VRA elements.
 
This should be fixed now.

**Comment by aviannamiller** [comment]
Great! I am not getting any more of those "stuck" toggle buttons.

The new group announcements are a great improvement across screen readers, although it is a bit hard to test in entirety with NVDA due to that other issue with the extraneous read out; activating the toggle on groups with subgroups prompts a read out of the subgroup labels instead, while the appropriate announcement of group labels happens when toggling a group with fields instead of subgroups. Let's definitely revisit after merging.

**Comment by kimisgold** [comment]
Now that the drawer toggle pattern branch has been merged into the core, I think it's a good time to revisit the issues we were having with extraneous screen reader announcements. Could you run through the toggles in this plugin again @aviannamiller?

**Comment by aviannamiller** [comment]
Things look great. I am no longer seeing any of the previously reported issues. This should be good to close!
